src/dmd/expressionsem.d

```diff
@@ -598,8 +598,96 @@ private bool functionParameters(Loc loc, Scope* sc, TypeFunction tf, Type tthis,
                 }
                 arg = p.defaultArg;
                 arg = inlineCopy(arg, sc);
-                // __FILE__, __LINE__, __MODULE__, __FUNCTION__, and __PRETTY_FUNCTION__
-                arg = arg.resolveLoc(loc, sc);
+
+                if(auto argexp = arg.isArgStringInitExp())
+                {
+                  if(argexp.ident is null)
+                  {
+                    error(argexp.loc, "argexp.ident null");
+                    return true;
+                  }
+                  //assert(nargs > 0);
+                  bool found = false;
+
+                  TypeFunction orig = cast(TypeFunction)fd.originalType;
+                  TypeFunction temp2 = cast(TypeFunction)fd.type;
+                  assert(orig);
+                  string[]values;
+                  for(int u=0, u2=0; u<orig.parameters.dim; u++)
+                  {
+                    Parameter pu = Parameter.getNth(orig.parameters, u);
+                    bool is_variadic_param=true;
+                    int u2_begin=u2;
+                    for(;u2<fd.parameters.dim; u2++)
+                    {
+                        auto ident2 = (*fd.parameters)[u2].ident;
+                        if(ident2==pu.ident)
+                        {
+                            u2++;
+                            is_variadic_param=false;
+                            break;
+                        }
+                        else
+                        {
+                            // TODO: factor with where this magic string is defined
+                            enum vararg_name = "_param_";
+                            // TODO: define/use startsWith
+                            auto ident2b=ident2.toString;
+                            if(ident2b.length>vararg_name.length && ident2b[0..vararg_name.length]==vararg_name)
```

Comment:
Global comment: Please use a space before and after operators. Please format code like in the rest of the front end.